sig-and-wg/sig-core/proposals/introducing_knative_brokers.md

```diff
@@ -0,0 +1,90 @@
+
+1. For every new `Application` the `Application Operator` creates an `Event Service` and its corresponding  `Knative Channel` that will be used to fanout events to all allowed brokers.
+2. For every new namespace, Kyma Console ensures it has the label `knative-eventing-injection=enabled` so that a default `Broker` gets created by the Knative Eventing controller.
+3. Once the `Application Broker` gets a provision request for a `Service Instance` which has `bindable` parameter set to `false`, it will create a `Knative Subscription` that has:
+   - The channel pointing to the relevant `Event Service` fanout `Channel`
+   - The subscriber pointing to the default `Broker` of the namespace
+   - The ownerRef set as the `Service Instance` that just got provisioned which ensures the subscription gets deleted by Kubernetes garbage collector when the `Service Instance` gets deprovisioned.
+4. `Kyma Console` creates the proper `Trigger` objects for the `Lambda` function for every event type chosen as a lambda trigger.
+5. Events received by `Event Service` will be delivered to its channel which in turn will do a fanout to all connected brokers which will finally deliverd it to the lambda functions.
+
+### Solution 2: Coexisting
+
+<!-- markdownlint-disable MD024 -->
+#### Overview
+
+To offer Kyma Event Bus users (whether the UI or CRD users) a smoother transition and to better manage their expectations, it is possible to roll out the new feature of using `Knative Broker` a new `Beta` feature that the user can choose per namespaces to opt-in (referenced here as `opted-in namespaces`). Other namespaces (referenced here as `legacy namespaces`) would be using the old Event Bus.
+
+#### Architecture
+
+![architecture](assets/knative_brokers_co_exist_arch.svg)
+
+1. For every new `Application` the `Application Operator` creates an `Event Service` and its corresponding  `Knative Channel` that will be used to fanout events to the opted-in namespaces.
+2. From Kyma Console UI, when the user creates a namespace, the pop up will show an info message and toggle with something like:
+>Kyma now comes with a new `Knative Broker` based eventing feature (beta).
+>Would you like to try it out for this namespace?. Read more about it here <link to more docs>
+and by enabling the toggle, Kyma Console ensures it has the label `knative-eventing-injection=enabled` so that a default `Broker` gets created by the Knative Eventing controller.
+3. Once the `Application Broker` gets a provision request for a `Service Instance` which has `bindable` parameter set to `false`, it creates a `Knative Subscription` for `opted-in` namespaces (as in solution 1) or an `EventActivation` for `legacy namespaces`.
+4. `Kyma Console` creates the appropriate `Trigger` or `Kyma Subscription` objects for the `Lambda` function for `opted-in` and `legacy` namespaces.
+5. Publishing events from `Publish` will deliver events to subscribers in both `opted-in` and `legacy` namespaces.
```

Comment:
I have a few concerns with proposal 2:
- Publish service now needs to be a controller and needs to keep a state of all the new knative channels(for all applications) created to fanout the events. This is something I would avoid.
- What will be the response when one of the publish fails? E.g. there are 2 channels for 2 applications and a channel(old one). Will we fail the publish request when 1 channel fails and rest succeed? What are the implications?
- There should be another migration plan for this approach to a purely new approach which should be here IMO.
- We are not eliminating Publish Service but adding more logic to it which is against our objective.